raiden/transfer/state_change.py

```diff
@@ -223,6 +224,42 @@ def from_dict(cls, data):
         )
 
 
+class ActionChannelSetFee(StateChange):
+    def __init__(self, canonical_identifier: CanonicalIdentifier, mediation_fee: FeeAmount):
+        self.canonical_identifier = canonical_identifier
+        self.mediation_fee = mediation_fee
+
+    def __repr__(self):
+        return f'<ActionChannelSetFee id:{self.canonical_identifier} fee:{self.mediation_fee}>'
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, ActionChannelSetFee) and
+            self.canonical_identifier == other.canonical_identifier and
+            self.mediation_fee == other.mediation_fee
+        )
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            'canonical_identifier': self.canonical_identifier,
+            'fee': str(self.mediation_fee),
+        }
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> 'ActionChannelSetFee':
+        return cls(
+            canonical_identifier=data['canonical_identifier'],
```

Comment:
Shouldn't you cast this to `CanonicalIdentifier` ? Also in the to_dict method?

Example from where this is used elsewhere: [source](https://github.com/raiden-network/raiden/blob/ffbfcee9405a95e6a64d2282dc174d4b020d68f5/raiden/transfer/state_change.py#L1163)

```
            canonical_identifier=CanonicalIdentifier(
                token_network_address=to_canonical_address(data['token_network_identifier']),
                channel_identifier=ChannelID(int(data['channel_identifier'])),
                chain_identifier=CHAIN_ID_UNSPECIFIED,
            ),
```

Reply:
The attached source is constructing a `CanonicalIdentifier` instance because that class expects a canonical identifier in the constructor but takes each member individually and stores it in `self`. In our case, the JSONSerializer will do that automatically by calling the `from_dict` on this instance.

Reply:
Aha I see. So this will also be part of Konrad's breaking changes for the Canonical Identifier? He will have to do this everywhere.